sdk/core/core-client/src/urlHelpers.ts

```diff
@@ -192,23 +192,39 @@ function calculateQueryParameters(
   return result;
 }
 
-function simpleParseQueryParams(queryString: string): Array<[string, string]> {
+function simpleParseQueryParams(queryString: string): Map<string, string | string[]> {
+  const result: Map<string, string | string[]> = new Map<string, string | string[]>();
   if (!queryString || queryString[0] !== "?") {
-    return [];
+    return result;
   }
 
   // remove the leading ?
   queryString = queryString.slice(1);
-
   const pairs = queryString.split("&");
 
-  return pairs.map((pair) => {
+  for (const pair of pairs) {
     const [name, value] = pair.split("=", 2);
-    return [name, value];
-  });
+    const existingValue = result.get(name);
+    if (existingValue) {
+      if (Array.isArray(existingValue)) {
+        result.set(name, [...existingValue, value]);
```

Comment:
the destructuring here feels unnecessary since we're the ones creating the source array, so we're incurring a lot of GC for little benefit, I think we can get away with

```suggestion
        existingValue.push(value);
```

Reply:
Done